### ISSUE TITLE
[Help] can we migrate from realm to watermelon in an existing app? 

### ISSUE THREAD

**Issue by praveens96** [Nozbe/WatermelonDB#1230]
We have an app in production which is heavily dependent on realmdb. 

- if we want to switch to watermelon db what to be done?

- I think of a migration function on app launch to move all data in realm to watermelon. 
- post that, if I want to drop realm is that ok? 
- in the subsequent releases if I remove the realm package will it cause any issues? 

please help!!

**Comment by jrounsav** [comment]
I'm not sure that this is the best place to request this type of help since the approach is largely dictated by the situation your project is in rather than the state of watermelon, but my team did make this migration.

Our approach was to run both DBs in tandem, in production, with watermelon usage hidden behind a feature flag.

Once all features were fully supported by watermelon we set that flag to 100% so that everyone with the app installed used melon instead of realm.

When we confirmed there were no issues we then removed realm from the code and released a new version of the app to the app stores with only watermelon

The process to rewrite the affected sections of the app took us approx 6 months, however the time and work required will be HIGHLY specific to your project's situation.
